types/priv_validator_test.go

```diff
@@ -253,3 +255,15 @@ func clipToMS(t time.Time) time.Time {
 	nano = (nano / million) * million
 	return time.Unix(0, nano).UTC()
 }
+
+// See Issue https://github.com/tendermint/tendermint/issues/1290
+func TestPrivValidatorFSSaveDoesNotCrash(t *testing.T) {
+	saveDir, err := ioutil.TempDir("", "priv-validator")
+	require.Nil(t, err, "should have created the tempdir")
+
+	defer os.RemoveAll(saveDir)
+
+	pPath := filepath.Join(saveDir, "out-file", "tf.json")
+	pv := &PrivValidatorFS{filePath: pPath}
+	pv.Save()
```

Comment:
While it is a regression test, the behaviour we want to ensure is that the file is saved even in nested, non-existing directory structure. Would rename the test accordingly and test that the file with the right content is created for extra verification.

Reply:
Gotcha, thanks for that @xla, in deed that adds extra verification. In the past I've had pushback whenever I've pushed up rigorous tests, but am glad to get support for such, from you so I'll implement it :)

Reply:
Great stuff @xla recommending the rigorous validation, me likey!